src/transformers/modeling_utils.py

```diff
@@ -416,6 +417,52 @@ def tie_weights(self):
         if output_embeddings is not None:
             self._tie_or_clone_weights(output_embeddings, self.get_input_embeddings())
 
+        if self.config.is_encoder_decoder and self.config.tie_encoder_decoder:
+            self._tie_or_clone_encoder_decoder_weights(self.decoder, self.encoder)
+
+    def _tie_or_clone_encoder_decoder_weights(self, encoder, decoder):
+        uninitialized_encoder_weights = []
```

Comment:
should this be a set? list+= set later on confused me.

Reply:
would also be nice to type hint List[str] to show that its not the actual tensors.